app/services/credits/applied_coupon_service.rb

```diff
@@ -21,7 +21,8 @@ def create
         amount_currency: applied_coupon.amount_currency,
       )
 
-      applied_coupon.mark_as_terminated! if applied_coupon.coupon.percentage? || credit_amount >= remaining_amount
+      applied_coupon.frequency_duration -= 1 if applied_coupon.recurring?
+      should_terminate_applied_coupon?(credit_amount) ? applied_coupon.mark_as_terminated! : save_frequency_duration
```

Comment:
I'm generally not a big fan of ternary condition since it's a bit hard to read, can we change a bit this part?

Reply:
Refactored a bit (apply to other comments as well). At first I was wandering that Assignment Branch Condition size would be too high so I had few more helper methods, but at the end it's not...